api/functions/utils/clean_json.py

```diff
@@ -0,0 +1,206 @@
+"""
+Given a json file with transcript information these tools can perform
+- manipulations including generating word lists or filtering output to
+- exclude english/punctuation.
+- Optionally provide the output json file name with -j
+
+Copyright: University of Queensland, 2019
+Contributors:
+              Josh Arnold - (The University of Queensland, 2017)
+              Ben Foley - (The University of Queensland, 2018)
+              Nicholas Lambourne - (The University of Queensland, 2019)
+              Harry Keightley - (The University of Queensland, 2022)
+"""
+
+import re
+from typing import Dict, List, Set, Tuple
+
+
+def clean_json_data(
+    json_data: List[Dict[str, str]],
+    punctuation_to_collapse_by: str = "",
+    punctuation_to_explode_by: str = "",
+    special_cases: set = {None},
+    translation_tags: set = {None},
+    remove_english: bool = False,
+) -> List[Dict[str, str]]:
+    """
+    Clean a list of utterances (Python dictionaries) based on the given parameters.
+    :param json_data: List of Python dictionaries, each must have a 'transcript' key-value.
```

Comment:
Nit: please add a TODO here to google-style these.

Reply:
Fixed in next revision